pkg/cmd/cli/cmd/newapp.go

```diff
@@ -219,13 +233,109 @@ func RunNewApplication(fullName string, f *clientcmd.Factory, out io.Writer, c *
 			}
 		}
 	}
-	if len(result.List.Items) > 0 {
+	switch {
+	case len(installing) == 1:
+		// TODO: should get this set on the config or up above
+		_, kclient, err := f.Clients()
+		if err != nil {
+			return err
+		}
+		jobInput := installing[0].Annotations[newcmd.GeneratedForJobFor]
+		return followInstallation(f, jobInput, installing[0], kclient, out)
+	case len(installing) > 1:
+		for i := range installing {
+			fmt.Fprintf(c.Out(), "Track installation of %s with '%s logs %s'.\n", installing[i].Name, fullName, installing[i].Name)
+		}
+	case len(result.List.Items) > 0:
 		fmt.Fprintf(c.Out(), "Run '%s %s' to view your app.\n", fullName, StatusRecommendedName)
 	}
+	return nil
+}
+
+func followInstallation(f *clientcmd.Factory, input string, pod *kapi.Pod, kclient kclient.Interface, out io.Writer) error {
+	fmt.Fprintf(out, "Installing %q with pod %q ...\n", input, pod.Name)
+
+	// we cannot retrieve logs until the pod is out of pending
+	// TODO: move this to the server side
+	podClient := kclient.Pods(pod.Namespace)
+	if err := wait.PollImmediate(500*time.Millisecond, 60*time.Second, installationStarted(podClient, pod.Name, kclient.Secrets(pod.Namespace))); err != nil {
+		return err
+	}
+
+	mapper, typer := f.Object()
+	opts := &kcmd.LogsOptions{
+		Namespace:   pod.Namespace,
+		ResourceArg: pod.Name,
+		Options: &kapi.PodLogOptions{
+			Follow:    true,
+			Container: pod.Spec.Containers[0].Name,
+		},
+		Mapper:        mapper,
+		Typer:         typer,
+		ClientMapper:  f.ClientMapperForCommand(),
+		LogsForObject: f.LogsForObject,
+		Out:           out,
+	}
+	_, logErr := opts.RunLog()
+
+	// status of the pod may take tens of seconds to propagate
+	if err := wait.PollImmediate(500*time.Millisecond, 30*time.Second, installationComplete(podClient, pod.Name, out)); err != nil {
+		if err == wait.ErrWaitTimeout {
+			if logErr != nil {
+				// output the log error if one occurred
+				err = logErr
+			} else {
+				err = fmt.Errorf("installation may not have completed, see logs for %q for more information", pod.Name)
+			}
+		}
+		return err
+	}
 
 	return nil
 }
 
+func installationStarted(c kclient.PodInterface, name string, s kclient.SecretsInterface) wait.ConditionFunc {
+	return func() (bool, error) {
+		pod, err := c.Get(name)
+		if err != nil {
+			return false, err
+		}
+		if pod.Status.Phase == kapi.PodPending {
+			return false, nil
+		}
+		// delete a secret named the same as the pod if it exists
```

Comment:
> we should allow you to delegate without temporarily creating a secret with your own token

Yes we should, but delegation is a medium-sized feature. Better ideas using the tools we have available now are welcome. In the case of a lost connection, `oc logout` will invalidate the token.